### PR TITLE
Use yetch and support AbortController

### PR DESCRIPTION
**Pull request by tchak**
This is an attempt at #111

### CODE REVIEW

**Comment by stefanpenner** [comment]
Just noting some stuff, just doing diligence.

yetch: 3,169 (gzipd)
whatwg-fetch: 2,904 (gzipd)

Size difference basically doesn't matter. Getting abort control well justifies that.

---

I skimmed the code for `yetch` it also appears good.

File: public/fastboot-fetch.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
+/* globals define FastBoot */
 (function() {
   define('fetch', ['exports'], function(self) {
-    var fetch = FastBoot.require('node-fetch');
-    self['default'] = fetch;
-    self['Headers'] = fetch.Headers;
-    self['Request'] = fetch.Request;
-    self['Response'] = fetch.Response;
+    var AbortControllerPolyfill = FastBoot.require('abortcontroller-polyfill/dist/cjs-ponyfill');
```
Comment (stefanpenner) [inline public/fastboot-fetch.js]: Do we need our Fastboot users to also additionally import/require the `abortcontroller` polyfil, or does this happen automagically? If so, can you briefly explain how?

**Comment by stefanpenner** [comment]
cc @nlfurniss

**Comment by stefanpenner** [comment]
@tchak have you confirmed all the abort-controller stuff works correctly with node-fetch as well?

**Comment by jayphelps** [comment]
@stefanpenner @tchak yep node-fetch does not appear to support abort signals yet https://github.com/bitinn/node-fetch/pull/437

**Comment by stefanpenner** [comment]
@tchak I'm unsure if we should go down this path, until node-fetch supports signals. Thoughts?

File: public/fastboot-fetch.js (hunk 1 above)
Comment (tchak) [reply public/fastboot-fetch.js]: Just like with browser implementation :

```js
import fetch, { AbortController } from 'fetch';
```

I will add a paragraph to README to clarify

**Comment by tchak** [comment]
@stefanpenner regarding node support: https://github.com/mo/abortcontroller-polyfill#using-it-on-nodejs

**Comment by stefanpenner** [comment]
@tchak ah, thank-you.  This lgtm.

**Comment by stefanpenner** [comment]
released as v4.0.1 🎉

**Pull request merged**